packages/@guidesmiths/cuckoojs-schematics/src/husky/husky.factory.ts

```diff
@@ -0,0 +1,59 @@
+import type {
+	Rule,
+	SchematicContext,
+	Tree,
+} from '@angular-devkit/schematics';
+import {chain} from '@angular-devkit/schematics';
+import {normalize} from '@angular-devkit/core';
+import {execSync} from 'child_process';
+
+import {PackageJsonUtils} from '../utils/package-json.utils';
+
+export function main(options: {directory: string; skipInstall: boolean}): Rule {
+	return (tree: Tree, context: SchematicContext) => {
+		context.logger.info('Adding husky ...');
+		const path = normalize(options.directory);
+
+		// https://github.com/angular/angular-cli/blob/8da926966e9f414ceecf60b89acd475ce1b55fc5/packages/angular_devkit/schematics/src/tree/host-tree.ts#L332
+		if (!tree.getDir(normalize(`./${path}/.git`)).subdirs.length) {
+			context.logger.info(
+				'Git directory not found. Husky installation will not proceed. Please, consider initializing Git on this project',
+			);
+			return;
+		}
+
+		return chain([
+			updatePackageJson(path),
+			runCommand(path, options.skipInstall),
+		])(tree, context);
+	};
+}
+
+function runCommand(directory: string, skipInstall: boolean): Rule {
+	return (tree: Tree, context: SchematicContext) => {
+		if (skipInstall) {
+			context.logger.info(
+				'Husky schematic executed with skipInstall option. Husky will not be installed.',
+			);
+			return tree;
+		}
+
+		const path = `${directory}/.husky/commit-msg`;
+		execSync(`npm install --prefix=${directory}`);
+		execSync(`npx husky add ${path} 'npx --no -- commitlint --edit "$1"'`);
```

Comment:
This has a dependency on commitlint, so we need also need to install commitlint

Reply:
Do we? Maybe we should only install Husky and refer to the documentation to add that hook. This schematic is only intended to add Husky to the project. Not sure if it would be convenient to also add another dependency. Another approach would be to consider that Husky and Commitlint always go together for us